src/psyclone/transformations.py

```diff
@@ -2182,17 +2207,17 @@ def apply(self, outer):  # pylint: disable=arguments-differ
         # Move inner to parent:
         inner.parent = parent
         parent.children.insert(index, inner)
-        outer.children.remove(inner)
+        outer.loop_body.children.remove(inner)
 
-        # Move inner's children to outer:
-        for child in inner.children:
-            inner.children.remove(child)
-            outer.children.append(child)
-            child.parent = outer
+        # Move inner's schedule to outer:
+        outer.children[3] = inner.loop_body
```

Comment:
`loop_body` instead of `children[3]`?

Reply:
Not in this case as the loop_body setter is not implemented, and I think it shouldn't be as it should be always fixed as an Schedule. Does that make sense?

Reply:
Yes it does. Thanks for explaining.